jcli/src/jcli_app/rest/v0/message/mod.rs

```diff
@@ -62,9 +68,18 @@ fn post_message(file: Option<PathBuf>, addr: HostAddr, debug: DebugFlag) -> Resu
     let url = addr.with_segments(&["v0", "message"])?.into_url();
     let builder = reqwest::Client::new().post(url);
     let response = RestApiSender::new(builder, &debug)
-        .with_binary_body(msg_bin)
+        .with_binary_body(msg_bin.clone())
         .send()?;
     response.response().error_for_status_ref()?;
-    println!("Success!");
-    Ok(())
+
+    match Fragment::deserialize(msg_bin.into_buf()) {
```

Comment:
I think that `msg_bin` deserialization should be done before sending it. If it's malformed, what's the point in sending?